build_tools/buildkite/pipelines/benchmark/linux-benchmark.yml

```
@@ -0,0 +1,39 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+steps:
+  - label: ":hammer_and_wrench: Build"
+    key: "build"
+    commands:
+      - "docker run --user=$(id -u):$(id -g) --volume=\\${HOME?}:\\${HOME?} --volume=/etc/passwd:/etc/passwd:ro --volume=/etc/group:/etc/group:ro --volume=\\$PWD:\\$IREE_DOCKER_WORKDIR --workdir=\\$IREE_DOCKER_WORKDIR --rm gcr.io/iree-oss/frontends@sha256:5868b0de784a590d710c837c3799d9d52456510e6db3d555661ba5f9f87ce213 build_tools/cmake/build_linux_benchmark.sh"
+      - "tar --exclude='*.tar.gz' --exclude='*.tgz' --exclude='*.mlir' --exclude='*.tflite' -czvf benchmark-suites-${BUILDKITE_BUILD_NUMBER}.tgz build-host/benchmark_suites"
+      - "find build-host/benchmark_suites -name '*.mlir' | tar -czvf source-mlir-models-${BUILDKITE_BUILD_NUMBER}.tgz -T -"
+      - "tar -czvf iree-linux-x86_64-tools-${BUILDKITE_BUILD_NUMBER}.tgz build-linux-x86_64/tools/iree-benchmark-module build-linux-x86_64/tools/build_config.txt"
```

You can use a `|` here to just write a multiline string and no need for quotes. Some more documentation on the different multi-line yaml strings: https://yaml-multiline.info/

```suggestion
    commands: |
      docker run --user=$(id -u):$(id -g) --volume="\${HOME?}:\${HOME?}" --volume=/etc/passwd:/etc/passwd:ro --volume=/etc/group:/etc/group:ro --volume="\$PWD:\$IREE_DOCKER_WORKDIR" --workdir="\${IREE_DOCKER_WORKDIR}" --rm gcr.io/iree-oss/frontends@sha256:5868b0de784a590d710c837c3799d9d52456510e6db3d555661ba5f9f87ce213 build_tools/cmake/build_linux_benchmark.sh
      tar --exclude='*.tar.gz' --exclude='*.tgz' --exclude='*.mlir' --exclude='*.tflite' -czvf benchmark-suites-${BUILDKITE_BUILD_NUMBER}.tgz build-host/benchmark_suites
      find build-host/benchmark_suites -name '*.mlir' | tar -czvf source-mlir-models-${BUILDKITE_BUILD_NUMBER}.tgz -T -
      tar -czvf iree-linux-x86_64-tools-${BUILDKITE_BUILD_NUMBER}.tgz build-linux-x86_64/tools/iree-benchmark-module build-linux-x86_64/tools/build_config.txt
```

---

For some reason I thought the yaml multi-line strings would result in the command `cmd1 cmd2 cmd3` instead of `cmd1 && cmd2 && cmd3`. Thanks for pointing this out!

---

It results in

```shell
cmd1
cmd2
```

and buildkite effectively sets `set -e` so if any of those fail, the whole thing fails